main/main.cpp

```diff
@@ -1585,6 +1585,17 @@ Error Main::setup2(Thread::ID p_main_tid_override) {
 #endif
 		}
 
+		uint32_t boot_display_time = GLOBAL_DEF("application/boot_splash/display_time_msec", 0);
+		ProjectSettings::get_singleton()->set_custom_property_info("application/boot_splash/display_time_msec",
+				PropertyInfo(Variant::INT,
+						"application/boot_splash/display_time_msec",
+						PROPERTY_HINT_RANGE,
+						"0,100,1,or_greater")); // No negative numbers
+		if (boot_display_time > 0) {
+			// delay_msec() isn't yet available.
+			OS::get_singleton()->delay_usec(1000 * boot_display_time);
```

Comment:
This is definitely not the way it should work. You capture the time when the boot splash is set and then, later, at the end of setup2 when initialization completes (or more like, in start after loading the main scene) check how much time has elapsed and whether it was less than the time you configured, then if necessary wait.